InnerEye/ML/metrics.py

```diff
@@ -242,14 +242,23 @@ def calculate_metrics_per_class(segmentation: np.ndarray,
                          f"the label tensor indicates that there are {number_of_classes - 1} classes.")
     binaries = binaries_from_multi_label_array(segmentation, number_of_classes)
 
-    all_classes_are_binary = [is_binary_array(ground_truth[label_id]) for label_id in range(ground_truth.shape[0])]
-    if not np.all(all_classes_are_binary):
+    binary_classes = [is_binary_array(ground_truth[label_id]) for label_id in range(ground_truth.shape[0])]
+
+    # If ground truth image is nan, then will not be used for metrics computation.
+    nan_images = [np.isnan(np.sum(ground_truth[label_id])) for label_id in range(ground_truth.shape[0])]
+
+    # Validates if not binary then nan
+    assert np.all(np.array(binary_classes) == ~np.array(nan_images))
```

Comment:
you can feel a list into np.all, no need to convert to np.array. This check here would be a lot easier to understand if written inside the loop below. In fact, I'm not even sure this check does the right thing. Can you add tests for these special cases?

Reply:
More generally, this function has gotten a lot more complicated now, please add test coverage. It only operates on numpy arrays, so should be fairly easy to write them.